src/ApiCore/Transport/RestTransport.php

```diff
@@ -60,6 +68,30 @@ public function __construct(
         $this->httpHandler = $httpHandler;
     }
 
+    /**
+     * Builds a RestTransport.
+     *
+     * @param string $serviceAddress
+     *        The address of the API remote host, for example "example.googleapis.com".
+     * @param string $restConfigPath
+     *        Path to rest config file.
+     * @param array $config
```

Comment:
I understand if this would be considered outside the scope of this PR, so maybe we can file it as a follow up - but it would be great if we also supported proxying through HTTP client options.